indico/modules/events/registration/client/js/components/participant-list/types.ts

```diff
@@ -0,0 +1,54 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+export interface TableColumnObj {
+  id: number;
+  text: string;
+  is_picture: boolean;
+}
+
+export interface TableRowObj {
+  id: number;
+  checked_in: boolean;
+  columns: TableColumnObj[];
+}
+
+export interface TableObj {
+  headers: string[];
+  rows: TableRowObj[];
+  num_participants: number;
+  show_checkin: boolean;
+  title?: string;
+}
+
+export interface ParticipantListProps {
+  published: boolean;
+  merged: boolean;
+  totalParticipantCount: number;
+  tables: TableObj[];
+  title: string;
+  eventId: number;
+  preview?: string;
```

Comment:
Good catch! I was supposed to change that at some point

Reply:
Used enum so that we can have any string attached to it at any time, without changing every string in the frontend when changes in backend. Unlikely but maybe good practice